cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -518,13 +519,6 @@ var (
 					)
 				}
 			}
-			if withClaimAuthenticationCode, _ := cmd.Flags().GetBool("with-claim-authentication-code"); withClaimAuthenticationCode {
-				device.ClaimAuthenticationCode = &ttnpb.EndDeviceAuthenticationCode{
-					Value: strings.ToUpper(hex.EncodeToString(random.Bytes(4))),
-				}
-				paths = append(paths, "claim_authentication_code")
-			}
-
 			claimOnExternalJS := len(device.ClaimAuthenticationCode.GetValue()) > 0
```

Comment:
For now, this should include a condition that this CAC was not randomly generated just above. So claiming on an external JS is mutually exclusive with the deprecated CAC generation.